docs/user-guide/concepts/parsers.md

```diff
@@ -0,0 +1,22 @@
+# Parsers
+
+A `parser` is capable of understanding more deeply a resource and expose
+that information via events so rules can be built on top of this information.
+E.g.: a `JavaScript` parser built on top of `ESLint` so rules for analyzing
+`JavaScript` files can be built.
+
+You can specify what `parser`s you want to use via the `.sonarwhalrc`
+configuration file:
+
+```json
+{
+    "parsers": ["parser1", "parser2"]
+}
+```
+
+## List of official `parser`s
+
+The built-in `parser`s are:
+
+* `javascript`: A `JavaScript` parser built on top of `ESLint` so rules for
```

Comment:
We need to add more information on how a rule can subscribe to this events and create rules based on it.

Reply:
Do you have some rule in mind? Can it be addressed in another PR?

Reply:
Documentation on what's exposed via the parser and how to use it should be in this PR IMO.

Reply:
@molant I have add something else to the documentation.